src/GridFS/ReadableStream.php

```diff
@@ -179,24 +180,16 @@ public function readBytes(int $length): string
 
     /**
      * Rename all revisions of the file.
-     *
-     * @return int The number of revisions renamed
      */
-    public function rename(string $newFilename): int
+    public function rename(string $newFilename): bool
     {
-        /** @var Document[] $revisions */
-        $revisions = $this->collectionWrapper->findFiles(
-            ['filename' => $this->file->filename],
-            ['typeMap' => ['root' => 'bson'], 'projection' => ['_id' => 1]],
-        );
-        $count = 0;
-
-        foreach ($revisions as $revision) {
-            $this->collectionWrapper->updateFilenameForId($revision->get('_id'), $newFilename);
-            $count++;
+        if (! isset($this->file->filename) || ! is_string($this->file->filename)) {
+            throw new LogicException('Cannot rename file without a filename');
         }
 
-        return $count;
+        $result = $this->collectionWrapper->updateFilenameForFilename($this->file->filename, $newFilename);
+
+        return $result->isAcknowledged();
```

Comment:
This would typically be `true`, unless a user specified a `writeConcern` of 0. In other cases there would be an exception. You can keep the previous return value by returning `$result->getModifiedCount()` (which can be `null` in case of a `writeConcern` of 0), or always return true as the exception will tell people if it didn't succeed.